core/column/column.service.js

```diff
@@ -101,9 +102,24 @@ export function collapse(view) {
 
 export function widthFactory(table, form) {
 	const layout = table.model.layout;
+	const columns = table.data.columns();
+	const columnMap = map(columns);
 	form = form || layout().columns;
 
+	const occupied = columns
+		.filter(c => form.hasOwnProperty(c.key) || ('' + c.width).indexOf('%') < 0)
```

Comment:
can we check this inside of reduce? Looks like we don't need filtered array itself